File: protobuf/protobuf3/CSharp/Symbol.cs
using System.Collections.Generic;

public class Symbol {
    public string Name { get; set; }
    public Symbol Type { get; set; }
    public TypeClassification Classification { get; set; }
    public Dictionary<string, Symbol> Members { get; } = new();
    public Symbol Parent { get; set; }

    public override string ToString()
    {
        var result = Name;
        var classification = Classification.ToString();
        result += " (with classification " + classification;
        if (Type != null) result += ", type " + Type.ToString();
        result += ")";
        if (Parent != null) result += " of " + Parent.ToString();
        return result;
    }
}

File: protobuf/protobuf3/Java/Symbol.java
import java.util.HashMap;
import java.util.Map;

public class Symbol {
    private String name;
    private TypeClassification classification;
    private final Map<String, Symbol> members = new HashMap<>();
    private Symbol parent;

    public Symbol()
    {
    }
    
    public Symbol(String name, TypeClassification classification)
    {
        this.name = name;
        this.classification = classification;
    }
    
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public TypeClassification getClassification() {
        return classification;
    }

    public void setClassification(TypeClassification classification) {
        this.classification = classification;
    }

    public Map<String, Symbol> getMembers() {
        return members;
    }

    public Symbol getParent() {
        return parent;
    }

    public void setParent(Symbol parent) {
        this.parent = parent;
    }

    @Override
    public String toString() {
        StringBuilder result = new StringBuilder();
        result.append(name);
        String classificationStr = classification != null ? classification.toString() : "null";
        result.append(" (with classification ").append(classificationStr);
        result.append(")");
        if (parent != null) {
            result.append(" of ").append(parent.toString());
        }
        return result.toString();
    }
}
